src/cpio/client_providers/kms_client_provider/azure/azure_kms_client_provider.cc

```diff
@@ -211,35 +207,20 @@ void AzureKmsClientProvider::GetSessionCredentialsCallbackToDecrypt(
 
     // Get test PEM private key and convert it to EVP_PKEY*
     auto privateKeyPem = GetTestPemPrivWrapKey();
-    privateKey = nullptr;
-    BIOWrapper bioWrapper(const_cast<BIO_METHOD*>(BIO_s_mem()));
-
-    // Get the BIO object from the wrapper
-    BIO* bio = bioWrapper.get();
-    if (bio == nullptr) {
-      char* error_string = ERR_error_string(ERR_get_error(), nullptr);
-      throw std::runtime_error(std::string("Failed to create BIO: ") +
-                               error_string);
-    }
-
-    BIO_write(bio, privateKeyPem.c_str(), privateKeyPem.size());
-
     // Add the constant to avoid the key detection precommit
```

Comment:
Can you create another PR to add detailed explanation for this like "We can't create the key on the fly because the attestation report...."?

Reply:
I documented the wrapping key concept in this PR. Have a look:)

Reply:
Looks great, thanks!